public/app/features/dashboard-scene/serialization/transformSceneToSaveModel.ts

```diff
@@ -245,3 +279,100 @@ export function dataLayersToAnnotations(layers: SceneDataLayerProvider[]) {
 
   return annotations;
 }
+
+export function sceneVariablesSetToVariables(set: SceneVariableSet) {
+  const variables: VariableModel[] = [];
+  for (const variable of set.state.variables) {
+    const commonProperties = {
+      name: variable.state.name,
+      label: variable.state.label,
+      description: variable.state.description,
+      skipUrlSync: Boolean(variable.state.skipUrlSync),
+      hide: variable.state.hide || VariableHide.dontHide,
+    };
+    if (variable instanceof QueryVariable) {
+      variables.push({
+        ...commonProperties,
+        current: {
+          // TODO
+          // selected: false
+          // @ts-expect-error
+          value: variable.state.value,
+          // @ts-expect-error
+          text: variable.state.text,
+        },
+        options: [],
+        query: variable.state.query,
+        datasource: variable.state.datasource,
+        sort: variable.state.sort,
+        refresh: variable.state.refresh,
+        regex: variable.state.regex,
+        allValue: variable.state.allValue,
+        includeAll: variable.state.includeAll,
+        multi: variable.state.isMulti,
+        skipUrlSync: Boolean(variable.state.skipUrlSync),
+        hide: variable.state.hide || VariableHide.dontHide,
+      });
+    } else if (variable instanceof CustomVariable) {
+      variables.push({
+        ...commonProperties,
+        current: {
+          // TODO
+          // selected: false
+          // @ts-expect-error
+          text: variable.state.value,
+          // @ts-expect-error
+          value: variable.state.value,
+        },
+        options: variable.state.options.map((option) => ({
+          text: option.label,
+          value: option.value.toString(),
+          selected: Array.isArray(variable.state.value)
+            ? Boolean(variable.state.value.find((v) => v === option.value))
+            : variable.state.value === option.value,
+        })),
```

Comment:
this dos not look correct, for custom variables only the query property should be persisted not the options

Reply:
I was as surprised as you are :)